libkineto/src/CuptiActivityProfiler.cpp

```diff
@@ -1269,23 +1302,24 @@ const time_point<system_clock> CuptiActivityProfiler::performRunLoopStep(
         VLOG_IF(1, currentIter > 0)
             << "This state change was invoked by application's step() call";
 
-        if (libkineto::api().client()) {
-          libkineto::api().client()->stop();
-        }
-
-#if defined(HAS_CUPTI) || defined(HAS_ROCTRACER)
-        if (cupti_.stopCollection) {
-          ecs_.cupti_stopped_early = cupti_.stopCollection;
-          LOG(ERROR)
-              << "State: CollectTrace stopped by CUPTI. (Buffer size configured is "
-              << config_->activitiesMaxGpuBufferSize() / 1024 / 1024 << "MB)";
+        // currentIter > 0 means this is an iteration-based collection,
+        // triggered by pytorch main thread, it should be executed in another
+        // thread in case pytorch main thread is blocked
+        if (currentIter > 0) {
```

Comment:
Good catch! I think this didn't get caught because all it would do is make a single step trace still synchronous but we should make it async for all traces. Thanks!